lib/faexport/scraper.rb

```diff
@@ -489,7 +489,228 @@ def submit_journal(title, description)
     }
   end
 
+  def new_submissions(from_id)
+    # Set pagination
+    url = "msg/submissions/new"
+    if from_id
+      url << "~#{from_id}@72/"
+    end
+
+    # Get page code
+    html = fetch(url)
+
+    login_user = get_current_user(html)
+    submissions = html.css('.gallery > figure').map{|art| build_submission_notification(art)}
+    {
+        "current_user": login_user,
+        "new_submissions": submissions
+    }
+  end
+
+  def notifications(include_deleted)
+    # Get page code
+    html = fetch("msg/others/")
+    # Parse page
+    login_user = get_current_user(html)
+    # Parse new watcher notifications
+    new_watches = []
+    if html.at_css("ul#watches")
+      html.at_css("ul#watches").css("li:not(.section-controls)").each do |elem|
```

Comment:
I'd assign `html.at_css("ul#watches")` to a variable and check it so it doesn't need to be searched for twice

Reply:
Sounds good, done that in 674321a65a03f75da8ffef5cfae457331202288e to all notification types